tooling/lsp/src/lib.rs

```diff
@@ -175,3 +189,29 @@ fn byte_span_to_range<'a, F: files::Files<'a> + ?Sized>(
         None
     }
 }
+
+pub(crate) fn resolve_workspace_for_source_path(
+    file_path: &Path,
+) -> Result<nargo::workspace::Workspace, LspError> {
+    let package_root = find_file_manifest(file_path);
+
+    let toml_path = match package_root {
```

Comment:
I believe this can be made more rusty by doing  `package_root .unwrap_or..something`

Reply:
Personally, match is more readable...